src/Models/TableItem.php

```diff
@@ -69,4 +71,61 @@ public function getCMSFields()
 
         return $fields;
     }
+
+    /**
+     * Get a row of cells
+     *
+     * Retrieve cells for this row in the table. Only the number of cells
+     * configured in the containing TableBlock will be returned.
+     *
+     * @return ArrayList
+     */
+    public function getCells()
+    {
+        $cells = [];
+        foreach (range(1, $this->TableBlock->NumberOfColumns) as $cell) {
+            $cells[] = DBField::create_field('HTMLText', $this->{"Cell" . $cell});
+        }
+        $cells = ArrayList::create($cells);
+        return $cells;
+    }
+
+    /**
+     * Check if the FirstColumnIsHeader is checked and is the first row
+     *
+     * @return Boolean
+     */
+    public function getFirstCellIsHeader($Position)
+    {
+        return $this->TableBlock->FirstColumnIsHeader == true && $Position == 1;
+    }
+
+    /**
+     * Get the header list if the FirstColumnIsHeader is checked
+     *
+     * @return ArrayList
+     */
+    public function getHeadingRow()
+    {
+        if ($this->TableBlock->FirstRowIsHeader == true) {
+            return $this->TableBlock->TableItems()->First()->getCells();
+        }
+    }
+
+    /**
+     * Get the width of the columns
+     *
+     * @return ArrayList
+     */
+    public function getColumnProportions()
+    {
+        $proportions = ArrayList::create();
+        $cols = $this->TableBlock->NumberOfColumns;
+        foreach (range(0, $cols) as $i) {
+            $key = 'PropCol' . $i;
+            $value = $this->TableBlock->$key;
+            $proportions[] = DBField::create_field('HTMLFragment', $value, $key);
+        }
+        return $proportions;
```

Comment:
There is no PropCol0,
In line 124, `range(0, $cols)` should start at 1, and line 127 should start `$proportions[$i] = `…

Reply:
It might be simpler to implement `getColumnProportions()` like so…

```php
    public function getColumnProportions($index)
    {
        $value = $this->TableBlock->{'PropCol' . $index};
        return DBField::create_field('Int', $value, 'PropCol' . $index);
    }
```

Then in the templates, you can remove the `.offsetGet`:

```html
        <th style="width: {$Top.ColumnProportions($Pos)}%;">
```

(I'm not sure that the `DBField::create_field()` call is needed either)